src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -1836,7 +1837,8 @@ internal ImmutableArray<DocumentId> GetRelatedDocumentIds(DocumentId documentId)
                 return ImmutableArray.Create<DocumentId>(documentId);
             }
 
-            return this.GetDocumentIdsWithFilePath(filePath);
+            var documentIds = this.GetDocumentIdsWithFilePath(filePath);
```

Comment:
same here. can we make sure these two method that always create new array being not in a hot call path.